pymove/core/pandas_discrete.py

```diff
@@ -0,0 +1,315 @@
+            Indicates name of column of place labels on symbolic trajectory.
+        mean_coef : float
+            Multiplication coefficient of the mean time for the segment, default 1.0
+        std_coef : float
+            Multiplication coefficient of sdt time for the segment, default 1.0
+        statistics : dataframe
+            Time Statistics of the pairwise local labels.
+        label_new_tid : String, optional(TID_PART by default)
+            The label of the column containing the ids of the formed segments.
+            Is the new splitted id.
+        drop_single_points : boolean, optional(True by default)
+            If set to True, drops the trajectories with only one point.
+        inplace : bool, optional, default True.
+            Represents whether the operation will be performed on
+            the data provided or in a copy.
+        Return
+        ------
+        PandasMoveDataFrame or None
+            Object with new features or None if ``inplace=True``.
+
+      """
+        if inplace:
+            data_ = self
+        else:
+            data_ = self.copy()
+        try:
+
+            if TIME_TO_PREV not in data_:
+                self.generate_dist_time_speed_features(TRAJ_ID)
+
+            if local_label not in data_:
+                raise ValueError('{} not in data frame.'.format(local_label))
+
+            if PREV_LOCAL not in data_:
+                data_[local_label] = data_[local_label].astype(np.float64)
+                self.generate_prev_local_features(label_id=label_id,
+                                                  local_label=local_label)
+
+            if statistics is None:
+                if (data_[PREV_LOCAL].isna().sum() == data_.shape[0]):
+                    raise ValueError('all values in the {} column are null.'
+                                     .format(PREV_LOCAL))
+                else:
+                    statistics = generate_time_statistics(data_, local_label=local_label)
+                    threshold_time_statistics(statistics, mean_coef, std_coef)
+
+            clean_trajectories_with_few_points(data_, label_tid=label_id,
+                                               min_points_per_trajectory=2, inplace=True)
+
+            current_tid, ids, count = _prepare_segmentation(data_, label_id, TID_STAT)
+
+            for idx in progress_bar(ids, desc='Generating %s' % TID_STAT):
+                md = data_.loc[idx, [TIME_TO_PREV, local_label, PREV_LOCAL]]
+                md = pd.DataFrame(md)
+
+                filter_ = []
+                for index, row in md.iterrows():
+                    local_label_ = row[local_label]
+                    prev_local = row[PREV_LOCAL]
+                    threshold = statistics[
+                        (statistics[local_label]
+                         == local_label_) & (statistics[PREV_LOCAL] == prev_local)
+                    ][THRESHOLD].values
+                    time_row = row[TIME_TO_PREV]
```

Comment:
Codacy found an issue: [local variable 'time_row' is assigned to but never used (F841)](https://app.codacy.com/gh/InsightLab/PyMove/pullRequest?prid=6293208)